### PR TITLE
Raven db 12860

### PR DESCRIPTION
**Pull request by talweiss1982**

### CODE REVIEW

File: src/Raven.Client/ServerWide/Commands/TcpConnectionInfo.cs [hunk 1]
```diff
@@ -7,15 +7,27 @@ public class TcpConnectionInfo
         public int Port;
         public string Url;
         public string Certificate;
-
+        public string[] ServerUrls;
```
Comment (ppekrol) [inline src/Raven.Client/ServerWide/Commands/TcpConnectionInfo.cs]: can we rename that to `Urls` the single Url property is called `Url`, do not see a good reason to not call this property `Urls`

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 2]
```diff
@@ -156,17 +157,49 @@ private Task Run(CancellationToken ct)
 
         private async Task<Stream> ConnectToServer(CancellationToken token)
         {
-            var command = new GetTcpInfoCommand("Subscription/" + _dbName, _dbName);
+            async Task<TcpConnectionInfo> LegacyTryGetTcpInfo(RequestExecutor requestExecutor1, JsonOperationContext context, ServerNode node = null)
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: `requestExecutor1` ?

Comment (ayende) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: And can we move this to a separate method?

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 3]
```diff
@@ -156,17 +157,49 @@ private Task Run(CancellationToken ct)
...
+            {
+                var tcpCommand = new GetTcpInfoCommand("Subscription/" + _dbName, _dbName);
+                try
+                {
+                    if(node != null)
+                    {
+                        await requestExecutor1.ExecuteAsync(node, null, context, tcpCommand, shouldRetry: false, sessionInfo: null, token: token)
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: why `shouldRetry`  is `false`?

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 4]
```diff
@@ -156,17 +157,49 @@ private Task Run(CancellationToken ct)
...
                 if (_redirectNode != null)
                 {
                     try
                     {
-                        await requestExecutor.ExecuteAsync(_redirectNode, null, context, command, shouldRetry: false, sessionInfo: null, token: token).ConfigureAwait(false);
+                        await requestExecutor.ExecuteAsync(_redirectNode, null, context, command, shouldRetry: false, sessionInfo: null, token: token)
+                            .ConfigureAwait(false);
+                        tcpInfo = command.Result;
+                    }
+                    catch (ClientVersionMismatchException)
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:175]: what would throw this error?

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs (hunk 2 above)
Comment (talweiss1982) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: resharper... sure

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs (hunk 3 above)
Comment (talweiss1982) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs]: I don't know this was the original logic, I think this is the case where you've already been redirected

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs (hunk 4 above)
Comment (talweiss1982) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:175]: the router when it doesn't find the path for the endpoint, I have tested this by modifying the remote task URL to something that doesn't exist and it propagate nicely to the client

**Comment by ppekrol** [comment]
https://github.com/ravendb/ravendb/pull/8948

**Pull request merged**